File: 9-PalindromeNumber/9-PalindromeNumber.py
# Last updated: 09/10/2025, 09:40:31
class Solution(object):
    def isPalindrome(self, x):
        """
        :type x: int
        :rtype: bool
        """
        if x <0:
            return False
        reverse = int(str(x)[::-1])
        if x == reverse:
            return True
        else:
            return False

        

File: 9-PalindromeNumber/9-PalindromeNumber.cpp
// Last updated: 03/03/2026, 20:07:36
class Solution {
public:
    bool isPalindrome(int x) {

        int n = x;
        if (n < 0)
            return false;
        
        vector <int> revDigits;
        while (n > 0){

            revDigits.push_back ((int)(n %10));
            n/=10;
        }

         long long result = 0;
        for (int d : revDigits) {
            result = result * 10 + d;
            if (result > INT_MAX) return 0;
        }

        if ( result == x){
            return true;
        }
        else{
            return false;
        }



    }
};